### ISSUE TITLE
Game crashing when almost done downloading

### ISSUE THREAD

**Issue by willc5** [Pokechu22/WorldDownloader#142]
Log: https://pastebin.com/1uDXWx9c

**Comment by Pokechu22** [comment]
I don't think this crash is directly related to WDL; it seems like you just ran out of threads (`java.lang.OutOfMemoryError: unable to create new native thread`) in code related to the sound manager(?).  This isn't [something that should normally happen](https://stackoverflow.com/a/16789621) so I'm not sure why it did.  Does it happen consistently?

**Comment by willc5** [comment]
Yeah it has happened every time I tried downloading worlds

**Comment by Pokechu22** [comment]
Do you also have optifine installed?  I note that your install path includes `MultiMC\instances\Opti`.  Optifine doesn't work with WDL as a base-edit, and having it installed might cause issues (and it probably wouldn't work anyways).  I don't *think* it'd cause this issue, but combining it causes weird things to happen so I wouldn't be too surprised.  I'd recommend either creating a separate instance or just removing optifine when using WDL and removing WDL when using optifine.

**Comment by willc5** [comment]
I did, but I disabled it because the game wouldn't start with it installed. It is still in there, just not enabled. I'll try making another instance with just WDL.

**Comment by willc5** [comment]
it doesn't seem to be crashing now but i ran out of memory this time lol

how do I give more memory? it says this
Error occurred during initialization of VM
Could not reserve enough space for 2097152KB object heap
Process exited with code 1.

**Comment by willc5** [comment]
Nvm, fixed the memory issue by downloading java64 and allocating more memory. But there are about 5 chunks that don't save when I stand in them (not showing in saved chunks)
http://prntscr.com/p97udd

**Comment by Pokechu22** [comment]
Are the chunks all void?  Those are *generally* skipped when saving (to avoid overwriting actual chunks, for the most part).  Otherwise, I'm not sure why that would happen, and it'd be worth checking the downloaded world itself to see if they're actually missing there or if it's an issue with the saved chunks GUI.
